analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeDeclarationSyntaxExtensionsTest.cs

```diff
@@ -91,5 +91,148 @@ private void Method2()
+        }
+
+        [TestMethod]
+        public void ParameterList_Interface()
+        {
+            var (tree, model) = TestHelper.CompileCS($$"""
+            interface Test
+            {
+            }
+            """);
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var parameterList = typeDeclaration.ParameterList();
+            parameterList.Should().BeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow(LanguageVersion.CSharp1)]
+        [DataRow(LanguageVersion.CSharp2)]
+        [DataRow(LanguageVersion.CSharp3)]
+        [DataRow(LanguageVersion.CSharp4)]
+        [DataRow(LanguageVersion.CSharp5)]
+        [DataRow(LanguageVersion.CSharp6)]
+        [DataRow(LanguageVersion.CSharp7)]
+        [DataRow(LanguageVersion.CSharp7_1)]
+        [DataRow(LanguageVersion.CSharp7_2)]
+        [DataRow(LanguageVersion.CSharp7_3)]
+        [DataRow(LanguageVersion.CSharp8)]
+        [DataRow(LanguageVersion.CSharp9)]
+        [DataRow(LanguageVersion.CSharp10)]
+        [DataRow(LanguageVersion.CSharp11)]
+        [DataRow(LanguageVersion.CSharp12)]
+        public void PrimaryConstructor_NoPrimaryConstructor(LanguageVersion languageVersion)
+        {
+            var tree = CSharpSyntaxTree.ParseText("""
+            public class Test
+            {
+            }
+            """, new CSharpParseOptions(languageVersion));
+            var compilation = CSharpCompilation.Create(assemblyName: null, syntaxTrees: new[] { tree });
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            typeDeclaration.PrimaryConstructor(compilation.GetSemanticModel(tree)).Should().BeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow(LanguageVersion.CSharp9)]
+        [DataRow(LanguageVersion.CSharp10)]
+        [DataRow(LanguageVersion.CSharp11)]
+        [DataRow(LanguageVersion.CSharp12)]
+        public void PrimaryConstructor_PrimaryConstructorRecord(LanguageVersion languageVersion)
+        {
+            var tree = CSharpSyntaxTree.ParseText("""
+            public record Test(int i)
+            {
+            }
+            """, new CSharpParseOptions(languageVersion));
+            var compilation = CSharpCompilation.Create(assemblyName: null, syntaxTrees: new[] { tree });
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(compilation.GetSemanticModel(tree));
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().ContainSingle();
+            methodSymbol.Parameters.Should().ContainSingle(p => p.Name == "i" && p.Type.SpecialType == SpecialType.System_Int32);
```

Comment:
I would keep only the second assertion

Reply:
They are not the same. The first checks, that there only one entry in the list and the second looks for the characteristics of the entry. I changed the assertions to make them better readable.